cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java

```diff
@@ -906,16 +906,16 @@ public void setCoordinator(Coordinator coordinator) {
         if (ClusterDescriptor.getInstance().getConfig().isUseAsyncServer()) {
           AsyncDataClient client =
               ClusterIoTDB.getInstance()
-                  .getClientProvider()
                   .getAsyncDataClient(node, ClusterConstant.getReadOperationTimeoutMS());
           result =
               SyncClientAdaptor.getUnregisteredMeasurements(
                   client, partitionGroup.getHeader(), seriesList);
         } else {
-          try (SyncDataClient syncDataClient =
-              ClusterIoTDB.getInstance()
-                  .getClientProvider()
-                  .getSyncDataClient(node, ClusterConstant.getReadOperationTimeoutMS())) {
+          SyncDataClient syncDataClient = null;
+          try {
+            syncDataClient =
+                ClusterIoTDB.getInstance()
+                    .getSyncDataClient(node, ClusterConstant.getReadOperationTimeoutMS());
             try {
```

Comment:
We may merge these two `try` as long as `syncDataClient =
                ClusterIoTDB.getInstance()
                    .getSyncDataClient(node, ClusterConstant.getReadOperationTimeoutMS())` doesn't throw `TException `.

There are so many similar places need be fixed as you abandon the `try-with-resource` grammer, you can refer to this [doc](https://shimo.im/docs/gWjV8HPJXJq9PdDr) to see why we need this judgement.